cesium_app/handlers/dataset.py

```diff
@@ -68,28 +57,31 @@ def post(self):
                                         str(uuid.uuid4()) + "_" +
                                         util.secure_filename(ts_path))
                            for ts_path in ts_paths]
+        d = Dataset(name=dataset_name, project=p, meta_features=meta_features)
         for old_path, new_path in zip(ts_paths, unique_ts_paths):
             os.rename(old_path, new_path)
-        file_names = [shorten_fname(ts_path) for ts_path in ts_paths]
-        d = Dataset.add(name=dataset_name, project=p, file_names=file_names,
-                        file_uris=unique_ts_paths, meta_features=meta_features)
+            d.files.append(DatasetFile(name=shorten_fname(old_path),
+                                       uri=new_path))
+        DBSession().add(d)
+        DBSession().commit()
 
         return self.success(d, 'cesium/FETCH_DATASETS')
 
     @tornado.web.authenticated
     def get(self, dataset_id=None):
         if dataset_id is not None:
-            dataset = self._get_dataset(dataset_id)
+            dataset = Dataset.get_if_owned_by(dataset_id, self.current_user)
             dataset_info = dataset.display_info()
         else:
-            datasets = [d for p in Project.all(self.current_user)
-                            for d in p.datasets]
+            datasets = [d for p in self.current_user.projects
+                        for d in p.datasets]
             dataset_info = [d.display_info() for d in datasets]
 
         return self.success(dataset_info)
 
     @tornado.web.authenticated
     def delete(self, dataset_id):
-        d = self._get_dataset(dataset_id)
-        d.delete_instance()
+        d = Dataset.get_if_owned_by(dataset_id, self.current_user)
```

Comment:
I think `get_if_owned_by` was a pre-existing name elsewhere; both parse the same to me, no strong opinion.